src/dex/nerve/nerve.ts

```diff
@@ -0,0 +1,399 @@
+    blockNumber: number,
+  ): Promise<string[]> {
+    return this.allPools
+      .filter(pool => {
+        return (
+          pool.tokenAddresses.includes(srcToken.address.toLowerCase()) &&
+          pool.tokenAddresses.includes(destToken.address.toLowerCase())
+        );
+      })
+      .map(pool => Nerve.getIdentifier(this.dexKey, pool.address));
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<NerveData>> {
+    try {
+      if (side === SwapSide.BUY) return null;
+
+      const _srcToken = wrapETH(srcToken, this.network);
+      const _destToken = wrapETH(destToken, this.network);
+
+      if (
+        _srcToken.address.toLowerCase() === _destToken.address.toLowerCase()
+      ) {
+        return null;
+      }
+
+      const filterPoolsByIdentifiers = (
+        identifiers: string[],
+        pools: EventPoolOrMetapool[],
+      ) => {
+        return pools.filter(pool =>
+          identifiers.includes(Nerve.getIdentifier(this.dexKey, pool.address)),
+        );
+      };
+
+      const selectedPools = !limitPools
+        ? filterPoolsByIdentifiers(
+            await this.getPoolIdentifiers(
+              _srcToken,
+              _destToken,
+              side,
+              blockNumber,
+            ),
+            this.allPools,
+          )
+        : filterPoolsByIdentifiers(limitPools, this.allPools);
+
+      const statePoolPair = await this.getStates(selectedPools, blockNumber);
+
+      // here side === SwapSide.SELL
+      const unitVolume = getBigIntPow(10 ** _srcToken.decimals);
+      const chunks = amounts.length - 1;
+
+      const _width = Math.floor(chunks / NERVE_CHUNKS);
+      const _amounts = [unitVolume].concat(
+        Array.from(Array(NERVE_CHUNKS).keys()).map(
+          i => amounts[(i + 1) * _width],
+        ),
+      );
+
+      const result: ExchangePrices<NerveData> = [];
+      for (const { pool, state } of statePoolPair) {
+        // To be able to update state property isValid if calculations receive error
+        const _state = _.cloneDeep(state) as PoolState;
+
+        const srcIndex = pool.tokens.findIndex(
+          token =>
+            token.address.toLowerCase() === _srcToken.address.toLowerCase(),
+        );
+        const destIndex = pool.tokens.findIndex(
+          token =>
+            token.address.toLowerCase() === _destToken.address.toLowerCase(),
+        );
+
+        const _rates: bigint[] = [];
+        for (const _amount of _amounts) {
+          const out = pool.math.calculateSwap(
+            _state,
+            srcIndex,
+            destIndex,
+            _amount,
+            // Actually we need here block timestamp, but +- 15 seconds shouldn't
+            // affect the calculations
+            bigIntify((Date.now() / 1000).toFixed(0)),
+          );
+          if (out === null) {
```

Comment:
I don't think `calculateSwap` returns `null` any more?  So you should catch the error.